File: ch8/yojkim/ch8/dollar.go
package ch8

type Dollar struct {
	amount int
}

func NewDollar(amount int) Dollar {
	return Dollar{amount}
}

func (d Dollar) Times(multiplier int) Money {
	return Dollar{d.amount * multiplier}
}

func (d Dollar) Equal(input Money) bool {
	dollar, ok := input.(Dollar)
	if !ok {
		return false
	}
	return d.amount == dollar.amount
}

File: ch7/gunddol/src/ch7/dollar.java
package ch7;

public class dollar extends Money{

    public dollar(int amount){
        this.amount = amount;
    }

    public dollar times(int multiplier) {
        return new dollar(amount * multiplier);
    }

    public boolean equals(Object object){
        Money money = (Money) object;
        return amount == money.amount;
    }
}
